test/detection/test_line_counter.py

```diff
@@ -70,3 +73,320 @@ def test_calculate_region_of_interest_limits(
+                [100, 50, 120, 70],
+            ],
+            [False, False, True],
+            [False, False, False],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(100, 100),
+            ),
+            [
+                [50, 45, 70, 30],
+                [40, 50, 50, 40],
+                [0, 50, 10, 40],
+            ],
+            [False, False, False],
+            [False, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(100, 0),
+            ),
+            [
+                [50, -45, 70, -30],
+                [40, 50, 50, 40],
+            ],
+            [False, False],
+            [False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, -100),
+            ),
+            [
+                [100, -50, 120, -70],
+                [-100, -50, -80, -70],
+            ],
+            [False, True],
+            [False, False],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(50, 100),
+            ),
+            [
+                [50, 50, 70, 30],
+                [40, 50, 50, 40],
+                [0, 50, 10, 40],
+            ],
+            [False, False, False],
+            [False, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+            ],
+            [False, False, True, False, True, False, True, False],
+            [False, True, False, True, False, True, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(-100, 0),
+            ),
+            [
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+            ],
+            [False, False, True, False, True, False, True, False],
+            [False, True, False, True, False, True, False, True],
+        ),
+    ],
+)
+def test_line_zone_single_detection(
+    vector, bbox_sequence, expected_count_in: List[bool], expected_count_out: List[bool]
+) -> None:
+    line_zone = LineZone(start=vector.start, end=vector.end)
+    for i, bbox in enumerate(bbox_sequence):
+        detections = mock_detections(
+            xyxy=[bbox],
+            tracker_id=[0],
+        )
+        count_in, count_out = line_zone.trigger(detections)
```

Comment:
Once again, please rename `count_in` and `count_out` to `crossed_in` and `crossed_out` respectively.